dts/bindings/pwm/nordic,nrf-sw-pwm.yaml

```diff
@@ -5,9 +5,9 @@ compatible: "nordic,nrf-sw-pwm"
 include: [pwm-controller.yaml, base.yaml]
 
 properties:
-    timer-instance:
-      type: int
-      description: Timer instance to use for generating the PWM output signals
+    generator:
+      type: phandle
+      description: Reference to TIMER instance for generating PWM output signals
```

Comment:
This will break compatibility with any code that's using timer-instance.

Do you think you could fall back on `timer-instance` if `generator` is not defined?

Something like this:

```c
#if DT_INST_NODE_HAS_PROP(0, timer_instance) && DT_INST_NODE_HAS_PROP(0, generator)
#error "Please define either the timer-instance or generator property, but not both"
#elif DT_INST_NODE_HAS_PROP(0, timer_instance)
#define GENERATOR_ADDR _CONCAT(NRF_TIMER, DT_INST_PROP(0, timer_instance))
#else
#define GENERATOR_ADDR	((NRF_TIMER_Type *) DT_REG_ADDR(GENERATOR_NODE))
#endif
```

If you can put in a comment saying 'timer-instance' is deprecated, that would be useful.

Reply:
Didn't we just get support for deprecated properties from #29417?  We could support both for a while, right?

If `timer-instance` is removed from the binding I wouldn't expect the conditional code to work.

Reply:
Sorry if I was unclear.

I am suggesting we do **not** remove this from the binding, exactly so the conditional code will work.

Reply:
I have updated code to keep `timer-instance` as deprecated property. Thanks for all suggestions!